src/compiler/core.ts

```diff
@@ -789,23 +789,17 @@ namespace ts {
      * @param comparer An optional `Comparer` used to sort entries before comparison, though the
      * result will remain in the original order in `array`.
      */
-    export function deduplicate<T>(array: ReadonlyArray<T>, equalityComparer?: EqualityComparer<T>, comparer?: Comparer<T>): T[];
-    export function deduplicate<T>(array: ReadonlyArray<T> | undefined, equalityComparer?: EqualityComparer<T>, comparer?: Comparer<T>): T[] | undefined;
-    export function deduplicate<T>(array: ReadonlyArray<T> | undefined, equalityComparer: EqualityComparer<T>, comparer?: Comparer<T>): T[] | undefined {
-        return !array ? undefined :
-            array.length === 0 ? [] :
+    export function deduplicate<T>(array: ReadonlyArray<T>, equalityComparer?: EqualityComparer<T>, comparer?: Comparer<T>): T[] {
+        return array.length === 0 ? [] :
             array.length === 1 ? array.slice() :
-            comparer ? deduplicateRelational(array, equalityComparer, comparer) :
+            comparer ? deduplicateRelational(array, equalityComparer!, comparer) :

```

Comment:
Do we pass undefined for `equalityComparer` to `deduplicate` anywhere? It seems like something we should verify since we're changing the signatures.

Reply:
It appears that `deduplicateRelational` will throw if `equalityComparer` is `undefined`: https://github.com/Microsoft/TypeScript/pull/26589/files#diff-66cec844e251a1918deb897eabca206bR767